tests/pytests/unit/modules/test_yumpkg.py

```diff
@@ -1,18 +1,42 @@
+import configparser
 import logging
 import os
+from functools import wraps
 
 import pytest
 
 import salt.modules.cmdmod as cmdmod
 import salt.modules.pkg_resource as pkg_resource
 import salt.modules.rpm_lowpkg as rpm
-import salt.modules.yumpkg as yumpkg
 import salt.utils.platform
-from salt.exceptions import CommandExecutionError, SaltInvocationError
+from salt.exceptions import CommandExecutionError, MinionError, SaltInvocationError
 from tests.support.mock import MagicMock, Mock, call, patch
 
 log = logging.getLogger(__name__)
 
+pytestmark = [
+    pytest.mark.skip_unless_on_linux,
+]
+
+
+# https://dev.to/stack-labs/how-to-mock-a-decorator-in-python-55jc
+def mock_decorator(*args, **kwargs):
+    """Decorate by doing nothing."""
+
+    def decorator(f):
+        @wraps(f)
+        def decorated_function(*args, **kwargs):
+            return f(*args, **kwargs)
+
+        return decorated_function
+
+    return decorator
+
+
+patch("salt.utils.decorators.path.which", mock_decorator).start()
```

Comment:
Unless you're testing, this is a **BIG NO**

Never patch at the global scope, or start a patch at the global scope

Reply:
Is there another way to patch a decorator?

Reply:
I removed the decorator from the download function and wrote a new test to test that functionality.